### PR TITLE
fix(streaming): find and return root actor failure when injection failed

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

After a source executor fails, the barrier manager may first realize that by finding the rx side is closed when injecting barriers, instead of receiving the error reported by the actor. This is because the barrier channel rx is immediately dropped when an executor exits, while error reporting happens later.

This PR makes it also call `try_find_root_failure` on barrier injection failure, making the error reporting more accurate.

Should address https://github.com/risingwavelabs/risingwave/issues/17389. The error message now looks like:

```
ERROR:  Failed to run the query

Caused by these errors (recent errors listed first):
  1: gRPC request to meta service failed: Internal error
  2: get error from control stream:
worker node 1, gRPC request to stream service failed:
Internal error:
failed to inject barrier: 
Actor 1 exited unexpectedly: 
Executor error: 
Connector error: 
failed to start cdc connector;
```

Still considering how to add a test to enforce it.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/task/barrier_manager.rs [hunk 1]
```diff
@@ -705,12 +707,15 @@ impl LocalBarrierWorker {
         assert!(!to_collect.is_empty());
 
         for actor_id in &to_collect {
-            if let Some(e) = self.failure_actors.get(actor_id) {
+            if self.failure_actors.contains_key(actor_id) {
                 // The failure actors could exit before the barrier is issued, while their
                 // up-downstream actors could be stuck somehow. Return error directly to trigger the
                 // recovery.
-                // try_find_root_failure is not used merely because it requires async.
```
Comment (BugenZhao) [inline src/stream/src/task/barrier_manager.rs:712]: This is the main issue being addressed.

File: src/stream/src/task/barrier_manager.rs [hunk 2]
```diff
@@ -431,14 +426,16 @@ impl LocalBarrierWorker {
                 }
                 completed_epoch = self.state.next_completed_epoch() => {
                     let result = self.on_epoch_completed(completed_epoch);
-                    self.control_stream_handle.inspect_result(result);
+                    if let Err(err) = result {
+                        self.notify_other_failure(err, "failed to complete epoch").await;
```
Comment (BugenZhao) [inline src/stream/src/task/barrier_manager.rs:430]: Not that sure when this branch will be hit and whether it's also necessary to collect actor failures.

**Comment by BugenZhao** [comment]
> Still considering how to add a test to enforce it.

cc @StrikeW Kindly ask for insights.

**Comment by BugenZhao** [comment]
* **#17685** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17685?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17672** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17672?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BugenZhao and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by StrikeW — APPROVED** [review]
LGTM

**Pull request merged**